HACKING.md

```diff
@@ -1,10 +1,88 @@
-The `manifests/` directory contains OpenShift manifests that will set up a Jenkins pipeline
-using the [Kubernetes Jenkins plugin](https://github.com/jenkinsci/kubernetes-plugin).
+The `manifests/pipeline.yaml` file contains an OpenShift template that
+will set up a Jenkins pipeline using the
+[Kubernetes Jenkins plugin](https://github.com/jenkinsci/kubernetes-plugin).
 
-### Create a Jenkins instance with a persistent volume backing store:
+These instructions will cover both local developer workflows, as well as
+interacting with the production pipeline in CentOS CI. A section header
+may indicate that the section only applies to one of the two workflows.
 
+### [LOCAL] Set up an OpenShift cluster
+
+First, make sure to install `oci-kvm-hook` on your host system (not in a
+pet container). This is required to ensure that the pipeline has access
+to `/dev/kvm`:
+
+```
+rpm-ostree install oci-kvm-hook # if on Silverblue
+dnf install -y oci-kvm-hook # if on traditional
+```
+
+We will use `oc cluster up` to set up a local cluster for testing. To do
+this, simply obtain the OpenShift binary from
+[the latest release](https://github.com/openshift/origin/releases/latest)
```

Comment:
any reason to not just tell them to install `origin-clients` rpm? and combine with the step above?

We could also mention if they don't want to use an rpm they can grab from github.

Reply:
Because then we'd need to make sure the instructions keep up with the interface for whatever version is in the repos right now. Today, that's 3.11, which looks very different from v3.9 (i.e. different switches). I also couldn't get 3.11 to work when trying to deploy v3.6.1 (I don't remember the exact error, but in general it's not clear if deploying any version of origin with any version of `origin-clients` more recent is definitely supported). If I or someone else gets it working with 3.11, then I'm totally up for recommending that and suggesting `origin-clients-3.11` as an alternative source.